wordpress-mobile/WordPress-Android
# Issue/12028 discover view holder

Parent issue #12028 

This PR initialises views on the reader post card based on ReaderPostUiState. 

I plan to replace [this method](https://github.com/wordpress-mobile/WordPress-Android/blob/588b9582c3ad022b91fd1be63111ac306d76c4ef/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L442) and start using the adapter there. But we are not there yet. I decided to create this PR to limit the size of this PR. The code isn't being used in production yet. However, you can test that at least the dummy/mocked discover post looks +- okay.

To test:
1. Make sure the reader improvements feature flag is on
2. Clear app data
3. Open reader
4. Select some interests and click on submit
5. Notice the discover post is being rendered correctly

- we'll run thorough tests when we start using this code for all the posts in the Reader. So don't worry about testing to much now.

PR submission checklist:
